cluster_management/src/main/java/com/pinterest/rocksplicator/task/DedupTask.java

```diff
@@ -0,0 +1,113 @@
+package com.pinterest.rocksplicator.task;
+
+import com.pinterest.rocksplicator.Utils;
+
+import org.apache.helix.task.Task;
+import org.apache.helix.task.TaskResult;
+import org.apache.helix.task.UserContentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DedupTask extends UserContentStore implements Task {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DedupTask.class);
+
+  private final String srcStorePathPrefix;
+  private final long resourceVersion;
+  private final String partitionName;
+  private final String taskCluster;
+  private final String job;
+  private final int adminPort;
+  private final String destStorePathPrefix;
+
+  public DedupTask(String srcStorePathPrefix, long resourceVersion, String partitionName,
+                   String taskCluster, String job, int adminPort,
+                   String destStorePathPrefix) {
+    this.srcStorePathPrefix = srcStorePathPrefix;
+    this.resourceVersion = resourceVersion;
+    this.partitionName = partitionName;
+    this.taskCluster = taskCluster;
+    this.job = job;
+    this.adminPort = adminPort;
+    this.destStorePathPrefix = destStorePathPrefix;
+  }
+
+  /**
+   * Execute the task.
+   * @return A {@link TaskResult} object indicating the status of the task and any additional
+   *         context information that can be interpreted by the specific {@link Task}
+   *         implementation.
+   */
+  @Override
+  public TaskResult run() {
+
+    if (srcStorePathPrefix.isEmpty() || resourceVersion == -1 || destStorePathPrefix.isEmpty()) {
+      String errMsg =
+          "Cancel the task, due to job command config map failed to provide all three of "
+              + "srcStorePathPrefix, resourceVersion, and destStorePathPrefix";
```

Comment:
What does `resourceVersion` correspond to?
Correct me if I missed something, but looks like resourceVersion is used only to construct the source and dest paths. Can we remove this logic from the task and let the client handle it? The client needs to pass in the path considering the version. This is a small detail but will make this task more generic and flexible.

Reply:
I would like to keep it for 2 reasons: 
1. keep consistent as backupTask. In backupTask, we made it more inclusive to parse version: client can pass a resourceVersion (i.e jobSubmission timestamp), or if not, we use the helixJobCreation time as the default version. Thus, the storage path in backup is: store_prefix/version/dbName. 
Therefore, to be consistent, we keep dedup has same format of : store_prefix/version/dbName. 
2.pass jobSubmission timestamp into Task will enable us to extract useful delay stats: taskExecutionTime - jobCreationTime/jobSubmissionTime. I think it might be useful to keep this optional open.